### ISSUE TITLE
Tracking integration for image similarity

### ISSUE THREAD

**Issue by sayakpaul** [huggingface/huggingface.js#338]
## Tracking integration of task - **image similarity**

Given that we have a [tutorial notebook on image similarity](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/image_similarity.ipynb) and an [upcoming blog post](https://github.com/huggingface/blog/pull/663), and given the usefulness of the use case, it's time we added support for this task. 

Note that you're not expected to do all of the following steps. This PR helps track all the steps required to get a new task fully supported in the Hub 🔥

- [ ] Integration with Inference API. Select at least one of the following:
  - [ ] Added a `transformers` `pipeline`
  - [ ] Added to Community Inference API for 3rd party library
  - [ ] Added to Community Inference API for generic
- [ ] Added basic UI elements (icon, order specification, etc)
- [ ] Added a widget

Integration guide: https://hf.co/docs/hub/models-tasks

**Comment by osanseviero** [comment]
Blocked by https://github.com/huggingface/transformers/issues/20830. Closing for now
